openfe/setup/methods/openmm/equil_rbfe_methods.py

```diff
@@ -0,0 +1,711 @@
+            softcore_electrostatics_alpha=alchem_settings.softcore_electrostatics_alpha,
+            softcore_sigma_Q=alchem_settings.softcore_sigma_Q,
+            interpolate_old_and_new_14s=alchem_settings.interpolate_old_and_new_14s,
+            flatten_torsions=alchem_settings.flatten_torsions,
+        )
+        hybrid_factory.addForce(
+            openmm.MonteCarloBarostat(
+                self._settings.barostat_settings.pressure.to(unit.bar).m,
+                self._settings.integrator_settings.temperature.m,
+                self._settings.barostat_settings.frequency.m,
+            )
+        )
+
+        lambdas = _rbfe_utils.lambdaprotocol.LambdaProtocol(
+            functions=alchem_settings.lambda_functions,
+            windows=alchem_settings.lambda_windows
+        )
+
+        selection_indices = hybrid_factory.hybrid_topology.select('all')
+        reporter = multistate.MultiStateReporter(
+            self._settings.simulation_settings.output_filename,
+            analysis_particle_indices=selection_indices,
+            checkpoint_interval=self._settings.simulation_settings.checkpoint_interval.m,
+        )
+
+        # Get platform and context caches
+        platform = _rbfe_utils.compute.get_openmm_platform(
+            self._settings.engine_settings.compute_platform
+        )
+        energy_context_cache = openmmtools.cache.ContextCache(
+            capacity=None, time_to_live=None, platform=platform,
+        )
+        sampler_context_cache = openmmtools.cache.ContextCache(
+            capacity=None, time_to_live=None, platform=platform,
+        )
+
+        int_settings = self._settings.integrator_settings
+        integrator = openmmtools.mcmc.LangevinSplittingDynamicsMove(
+            timestep=int_settings.timestep,
+            collision_rate=int_settings.collision_rate,
+            n_steps=int_settings.n_steps.m,
+            reassign_velocities=int_settings.reassign_velocities,
+            n_restart_attempts=int_settings.n_restart_attempts,
+            constraint_tolerance=int_settings.constraint_tolerance,
+        )
+
+        sampler = _rbfe_utils.multistate.HybridRepexSampler(
+            mcmc_moves=integrator,
+            hybrid_factory=hybrid_factory,
+        )
+        sampler.setup(
+            reporter=reporter,
+            platform=platform,
+            lambda_protocol=lambdas,
+            temperature=self._settings.integrator_settings.temperature,
+            endstates=alchem_settings.unsampled_endstates,
+        )
+        sampler.energy_context_cache = energy_context_cache
+        sampler.sampler_context_cache = sampler_context_cache
+
+        return True
+
+    def is_complete(self) -> bool:
+        results_file = self._settings.simulation_settings.output_filename
+
+        # TODO: Can improve upon this by checking expected length of the
+        #       nc archive?
+        return os.path.exists(results_file)
```

Comment:
This won't work long term, the file gets created on sampler creation.

Reply:
So this is currently detecting if run has started but not finished.. If we can infer from the settings how full the .nc should get we can check the "tail" of it.  This is minor as we're currently not thinking about this being threadsafe yet (i.e. we've only got one python process for now).

Reply:
Very dirty but done.